### PR TITLE
Fix RTMPose likelihood computation

### PR DESCRIPTION
**Pull request by maximpavliv**
### Background

The likelihoods (confidences) produced by `SimCCPredictor` were not consistent with the expected behavior of SimCC decoding.
- In some experiments, **likelihoods exceeded 1**, because the predictor was returning raw model outputs (logits) instead of proper probabilities.
- When setting `apply_softmax=True`, likelihoods were extremely small (~1e-3). This is expected mathematically (softmax distributes probability mass across the entire SimCC space), but not useful as a visibility/confidence measure.
- When setting `normalize_outputs=True`, most likelihoods were exactly 1, because normalization simply divided by the maximum logit value.

This mismatch originated from how the raw SimCC representation vectors (logits over discretized x/y coordinates) were converted into coordinates and visibility scores.

See the [original MMPose predictor implementation](https://github.com/open-mmlab/mmpose/blob/759b39c13fea6ba094afc1fa932f51dc1b11cbf9/mmpose/codecs/simcc_label.py) for reference.

### Changes

This PR brings `SimCCPredictor` closer to the [original MMPose decoding logic](https://github.com/open-mmlab/mmpose/blob/759b39c13fea6ba094afc1fa932f51dc1b11cbf9/mmpose/codecs/simcc_label.py#L193-L198):

**1. Added parameters**
  - `sigma`: variance of the Gaussian used to generate SimCC labels during training. This controls the expected sharpness of the label distribution.
  - `decode_beta`: scaling factor applied together with `sigma` to logits before softmax. It acts like an _inverse temperature_ in softmax: sharpening peaked distributions (visible keypoints) and flattening diffuse ones (occluded keypoints).
  
  Together, `sigma * decode_beta` ensures that visibility/confidence is decoded consistently with the label distribution used in training.

**2. Likelihood computation**
- If `normalize_outputs=False` (which is the default): the raw logits are scaled by (`sigma * decode_beta`) before calling `get_simcc_maximum` with `apply_softmax=True`. This matches the MMPose decoding step and yields meaningful visibility/confidence scores in `[0,1]`.

**3. Default behavior**
- `apply_softmax` now defaults to `True` (to ensure proper probabilities rather than raw logits are returned).
- `normalize_outputs` still defaults to `False`.

### Outcome
- Likelihoods are now properly bounded in `[0,1]`.
- The values reflect both the sharpness of the SimCC distribution and the visibility of the keypoint, consistent with the SimCC paper and MMPose implementation.
- Users can still enable `normalize_outputs` for compatibility with previous behavior, but it is discouraged for real confidence estimation.

### CODE REVIEW

**Comment by MMathisLab** [comment]
cc @maximpavliv can you adress copilots suggestions?

File: deeplabcut/pose_estimation_pytorch/apis/videos.py [hunk 1]
```diff
@@ -194,45 +194,25 @@ def video_inference(
...
         shelf_writer.open()
-    
-    pose_progress = tqdm(video, desc="Pose")
-    predictions = []
-    for i, frame in enumerate(pose_progress):
-        result = pose_runner.inference(images=[frame])
-        predictions.extend(result)
-    
+
+    predictions = pose_runner.inference(images=tqdm(video), shelf_writer=shelf_writer)
```
[the remark this answers is not in the document]
Comment (maximpavliv) [reply deeplabcut/pose_estimation_pytorch/apis/videos.py]: The `inference()` method always accepts this parameter, [it is defined in the `inference()` method of the generic `InferenceRunner` class](https://github.com/DeepLabCut/DeepLabCut/blob/12f308359c6b58408c81bb3bcc86286910cb57a3/deeplabcut/pose_estimation_pytorch/runners/inference.py#L135).

[the remark this answers is not in the document]
Comment (maximpavliv) [reply deeplabcut/pose_estimation_pytorch/apis/videos.py]: The `tqdm` bar is always displayed correctly, the `for data in images` loop is executed inside the `inference()` method ([here](https://github.com/DeepLabCut/DeepLabCut/blob/12f308359c6b58408c81bb3bcc86286910cb57a3/deeplabcut/pose_estimation_pytorch/runners/inference.py#L174) or [here](https://github.com/DeepLabCut/DeepLabCut/blob/12f308359c6b58408c81bb3bcc86286910cb57a3/deeplabcut/pose_estimation_pytorch/runners/inference.py#L365)).

File: deeplabcut/pose_estimation_pytorch/models/predictors/sim_cc.py [hunk 2]
```diff
@@ -42,21 +42,33 @@ class SimCCPredictor(BasePredictor):
...
         normalize_outputs: bool = False,
+        apply_softmax: bool = True,
+        sigma: float | int | tuple[float, ...] = 6.0,
+        decode_beta: float = 150.0,
     ) -> None:
         super().__init__()
         self.simcc_split_ratio = simcc_split_ratio
-        self.apply_softmax = apply_softmax
         self.normalize_outputs = normalize_outputs
+        self.apply_softmax = apply_softmax
```
[the remark this answers is not in the document]
Comment (maximpavliv) [reply deeplabcut/pose_estimation_pytorch/models/predictors/sim_cc.py]: Okay, I've restored the `SimCCPredictor`'s constructor arguments order!

**Comment by maximpavliv** [comment]
> cc @maximpavliv can you adress copilots suggestions?

@MMathisLab yes, done!

**Review by AlexEMG — APPROVED** [review]
